mempool/queue.go

```diff
@@ -0,0 +1,123 @@
+package mempool
+
+import (
+	"container/heap"
+	"sync"
+)
+
+var _ heap.Interface = (*TxPriorityQueue)(nil)
+
+type Comparator func(Tx, Tx) bool
```

Comment:
Is the goal here to avoid equal transactions?

Reply:
to build a queue based on some comparing func that will be provided by the user. It could be: gas, priority, size, or whatever he wants